core/src/main/groovy/noe/rhel/server/httpd/HttpdBaseOS.groovy

```diff
@@ -0,0 +1,114 @@
+ */
+@Slf4j
+class HttpdBaseOS extends Httpd {
+  protected String serviceName
+  private FileStateVault fileStateVault
+
+  HttpdBaseOS(String basedir, version) {
+    super(basedir, version)
+    this.serviceName = "httpd"
+    this.basedir = "/etc/${serviceName}"
+    this.fileStateVault = new FileStateVault()
+
+    setDefault()
+    logDirs.each { JBFile.makeAccessible(new File(getServerRoot() + it)) }
+  }
+
+  void setDefault() {
+    super.setDefault()
+    this.configDirs = ['/conf', '/conf.d']
+    if (version >= new Version("2.4")) {
+      this.configDirs.add('/conf.modules.d')
+    }
+    this.modulesDir = this.basedir + "/modules"
+    this.logDirs = ['/logs']
+    // what path will be taken as relative for star/stop cmd
+    this.binPath = '/'
+    if (platform.isRHEL8() || platform.isRHEL7()) {
+      this.start = ['systemctl', 'start', serviceName]
+      this.stop = ['systemctl', 'stop', serviceName]
+    } else if (platform.isRHEL6()) {
+      this.start = ['service', serviceName, 'start']
+      this.stop = ['service', serviceName, 'stop']
+    } else {
+      String osIdentification = platform.getOsName() + platform.getOsVersion() + platform.getOsArch()
+      log.error("Currently unsupported RHEL version. " + osIdentification)
+      throw new RuntimeException("Unsupported RHEL system version. " +osIdentification)
+    }
+    this.apachectl = ['/usr/sbin/apachectl']
+    this.deploymentPath = "/var/www/html"
+    this.confDeploymentPath = this.basedir + '/conf.d'
+    this.cgiDeploymentPath = "/var/www/cgi-bin"
+    this.modClusterCacheDir = "/var/cache/mod_cluster"
+    this.opensslPath = 'openssl' // openssl is on $PATH on RHEL by OS installation
+    this.abPath = "/usr/bin/ab"
+    this.htdbmPath = "/usr/bin/htdbm"
+    this.htpasswdPath = "/usr/bin/htpasswd"
+    this.httxt2dbmPath = "/usr/bin/httxt2dbm"
+    this.logresolvePath = "/usr/bin/logresolve"
+
+  }
+
+  File getPidFile() {
+    return new File("/var/run/${serviceName}/${serviceName}.pid")
+  }
+
+  Integer extractPid() {
+    int pid
+    try {
+      String pidFromFile = JBFile.read(pidFile).trim().replaceAll('"','')
+      pid = Integer.valueOf(pidFromFile)
+    } catch (e) {
+      pid = super.extractPid()
+    }
+
+    return pid
+  }
+
+  /**
+   * Stop the server
+   */
+  void start(Map conf = [:]) {
+    super.start(conf)
+    Library.letsSleep(5000)
+  }
+
+  /**
+   * Stop the server
+   */
+  long stop(Map conf = [:]) {
+    long stopTime = super.stop(conf)
+    Library.letsSleep(3000)
+    return stopTime
+  }
+
+  void setRequestedSELinuxContext(String context) {
+    this.start = ['runcon', '-t', context, this.start]
+  }
+
+  void backupConfs() {
+    configDirs.each { dir ->
+      File srcDir = new File(getServerRoot(), dir)
+      srcDir.listFiles().each { File conf ->
+        fileStateVault.push(conf)
+      }
+    }
+
+    configDirs.each { JBFile.makeAccessible(new File(getServerRoot() + it)) }
```

Comment:
Is it something what it is really needed ?

Reply:
Yes, it is for jenkins jobs to be able CRUD operations on files. Like workspace cleanup executed as "hudson" and if there is any content user hudson doesn't have privileges, it fails.

Reply:
Didn't you mention something about running under root privileges in comments above? That because of that we can safely work with `FileStateVault`? 

This seems to me like inconsistency, or do I missing something?

Reply:
You do not missing something, I had wrong code connection in my mind.
It is nothing we really need at all. Sorry for confusion.